packages/docusaurus-plugin-content-blog/src/__tests__/validation.test.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {PluginOptionSchema} from '../types';
+
+const defaultOptions = {
+  feedOptions: {},
+  beforeDefaultRehypePlugins: [],
+  beforeDefaultRemarkPlugins: [],
+  admonitions: {},
+  truncateMarker: /<!--\s*(truncate)\s*-->/,
+  rehypePlugins: [],
+  remarkPlugins: [],
+  showReadingTime: true,
+  blogTagsPostsComponent: '@theme/BlogTagsPostsPage',
+  blogTagsListComponent: '@theme/BlogTagsListPage',
+  blogPostComponent: '@theme/BlogPostPage',
+  blogListComponent: '@theme/BlogListPage',
+  postsPerPage: 10,
+  include: ['*.md', '*.mdx'],
+  routeBasePath: 'blog',
+  path: 'blog',
+};
+
+test('normalize options', () => {
+  const {value} = PluginOptionSchema.validate({});
+  expect(value).toEqual(defaultOptions);
+});
+
+test('validate options', () => {
+  const {value} = PluginOptionSchema.validate({
+    path: 'not_blog',
+    postsPerPage: 5,
+    include: ['api/*', 'docs/*'],
+    routeBasePath: 'not_blog',
+  });
+  expect(value).toEqual({
+    ...defaultOptions,
+    postsPerPage: 5,
+    include: ['api/*', 'docs/*'],
+    routeBasePath: 'not_blog',
+    path: 'not_blog',
+  });
+});
+
+test('throw Error in case of invalid options', () => {
+  const {error} = PluginOptionSchema.validate({
+    path: 'not_blog',
+    postsPerPage: -1,
+    include: ['api/*', 'docs/*'],
+    routeBasePath: 'not_blog',
+  });
+
+  expect(error).toMatchSnapshot();
+});
+
+test('throw Error in case of invalid feedtype', () => {
+  const {error} = PluginOptionSchema.validate({
+    feedOptions: {
+      type: 'none',
+    },
+  });
+
+  expect(error).toMatchSnapshot();
+});
+
+test('conversion of truncateMarker to Regex', () => {
+  const {value} = PluginOptionSchema.validate({
+    truncateMarker: 'tag',
+  });
+  expect(value).toEqual({
+    ...defaultOptions,
+    truncateMarker: /tag/,
+  });
```

Comment:
I guess this behavior is due to using `validate({convert: true})`.

Not sure it's something we want. I think it's safer to be the most strict possible by default and requiring the user to use a regex in the config.

I mostly use this conversion system for reading things like querystrings (as everything is converted to string).

I don't think either we should allow things like `postsPerPage: "10"`

If this becomes a problem, we can become less strict later. The opposite would be more annoying as it'd be a breaking change.

Reply:
I wrote the conversion for this one. Using a custom function.

Reply:
I thought that it was expected behavior. I will remove it

Reply:
I think if there was no conversion before, and nobody complained, we can do without that conversion :) this option is probably not widely used anyway

Reply:
Made is more strictier